### PR TITLE
translate/v2*tov3*: De-duplicate spec v2 passwdUsers

### PR DESCRIPTION
**Pull request by LorbusChris**
Similar functionality has been added to the MCO [1].
Adding deduplication for users here to keep implementations equivalent.

[1] openshift/machine-config-operator@ccb8d10

/cc @yuqi-zhang 
If there are no objections, I'll also open a PR to MCO to replace the function there with the one from this repo

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,6 +7,7 @@ require (
 	github.com/coreos/go-systemd v0.0.0-20181031085051-9002847aa142 // indirect
 	github.com/coreos/ignition v0.35.0
 	github.com/coreos/ignition/v2 v2.3.0
+	github.com/pkg/errors v0.9.1
```
Comment (bgilbert) [inline go.mod:10]: Can we split this into a separate commit?

File: translate/v23tov30/v23tov30.go [hunk 2]
```diff
@@ -518,6 +519,28 @@ func RemoveDuplicateFilesAndUnits(cfg old.Config) (old.Config, error) {
...
+	// Concat sshkey sections into the newest passwdUser in the list
+	// We make the assumption that there is only one user: core
+	// since that is the only supported user by design.
+	// It's technically possible, though, to have created another user
+	// during install time configs, since we only check the validity of
+	// the passwd section if it was changed. Explicitly error in that case.
+	if len(users) > 0 {
+		outUser := users[len(users)-1]
+		if outUser.Name != "core" {
+			return cfg, errors.Errorf("unexpected user with name: %v. Only the core user is supported.", outUser.Name)
```
Comment (bgilbert) [inline translate/v23tov30/v23tov30.go:531]: This is fine for the MCO but we shouldn't impose this restriction in ign-converter.

File: translate_test.go [hunk 3]
```diff
@@ -1281,7 +1281,7 @@ func TestTranslate3_1to2_4(t *testing.T) {
 	assert.Equal(t, exhaustiveConfig2_4, res)
 }
 
-func TestRemoveDuplicateFilesAndUnits2_3(t *testing.T) {
+func TestRemoveDuplicateFilesUnitsUsers2_3(t *testing.T) {
```
Comment (bgilbert) [inline translate_test.go:1284]: Let's also be sure to test the new functionality.

File: translate/v23tov30/v23tov30.go (hunk 2 above)
Comment (LorbusChris) [reply translate/v23tov30/v23tov30.go:531]: :+1: I'll update this part so we can merge arbitrary users and when switching to use this func in the MCO, the check can still be performed there.

**Comment by LorbusChris** [comment]
@bgilbert updated, PTAL :slightly_smiling_face:

File: translate_test.go [hunk 4]
```diff
@@ -1801,7 +1801,24 @@ func TestRemoveDuplicateFilesAndUnits2_3(t *testing.T) {
...
+			"two",
+		},
+	}
+	userTwo := types2_3.PasswdUser{
+		Name: userName,
+		SSHAuthorizedKeys: []types2_3.SSHAuthorizedKey{
+			"three",
+		},
+	}
+	testIgn2Config.Passwd.Users = append(testIgn2Config.Passwd.Users, userOne, userTwo)
```
Comment (bgilbert) [inline translate_test.go:1819]: It'd be good to add a third user with a different name and make sure it's kept separate in the output.

**Review by bgilbert — APPROVED** [review]

File: translate_test.go (hunk 4 above)
Comment (LorbusChris) [reply translate_test.go]: updated with a third user

**Review by bgilbert — APPROVED** [review]
